File: 0001_Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> record;
        for (int i = 0; i < nums.size(); i++) {
            if (record.find(target - nums[i]) != record.end())
                return {i, record[target - nums[i]]};
            record[nums[i]] = i;
        }
        return {};
    }
};

File: 0001_Two Sum/solution.go
func twoSum(nums []int, target int) []int {
    record := make(map[int]int)
    for i, s := range nums {
        v, e := record[target - s]
        if e == true {
            output := []int{v, i}
            return output
        }
        record[s] = i
    }
    return nil
}